src/AI/AIUtils.java

```diff
@@ -214,6 +216,36 @@ public static CommanderAbility queueCromulentAbility(Queue<GameAction> q, Comman
     }
     return retVal;
   }
+  
+  /**
+   * @return The area threatened by the unit, against the specified target type
+   */
+  public static Map<XYCoord, Double> findThreatPower(GameMap gameMap, Unit unit, UnitModel target)
+  {
+    XYCoord origin = new XYCoord(unit.x, unit.y);
+    Map<XYCoord, Double> shootableTiles = new HashMap<XYCoord, Double>();
+    ArrayList<XYCoord> destinations = Utils.findPossibleDestinations(unit, gameMap);
+    for( Weapon wep : unit.weapons )
+    {
+      if( wep.ammo > 0 )
```

Comment:
Should we also filter for whether `wep` can affect `target`? It doesn't look like there's a good reason to include spaces where the possible damage is zero.

Reply:
Fair enough